src/store/pipeline-report-sink.ts

```diff
@@ -53,16 +53,19 @@ export class PipelineReportSink implements ReportSink {
         log.info(`Saving report using ${name}...`);
         const maybeReportInfo = await sink.saveReport(lastReportInfo);
 
-        // Pass returned report info to the next sink
-        if (maybeReportInfo !== undefined) {
-          lastReportInfo = maybeReportInfo;
-          log.info(`Report saved using ${name}`, {
-            ...lastReportInfo,
-            report: undefined,
-          });
+        if (!maybeReportInfo) {
```

Comment:
`ReportSink` still returns `Promise<ReportInfo | undefined>`.
Maybe we should update it to return only `Promise<ReportInfo>` or throw and remove the `return` for both this `if` and `catch` so we don't terminate the loop.

Reply:
Ah, I see. I missed that `ReportSink` hadn't changed. I vote for simplifying that too (only return `Promise<ReportInfo>` or throw) and removing the return from `catch` to avoid altering the existing behavior.